src/stream/src/executor/top_n_new.rs

```diff
@@ -205,171 +205,47 @@ impl<S: StateStore> TopNExecutorBase for InnerTopNExecutorNew<S> {
         chunk: StreamChunk,
         epoch: u64,
     ) -> StreamExecutorResult<StreamChunk> {
-        let mut new_ops = vec![];
-        let mut new_rows = vec![];
+        let mut res_ops = vec![];
+        let mut res_rows = vec![];
```

Comment:
May initialize with capacity.

Reply:
Any fixes?

Reply:
There's some test cases still passing large limit into the top_n executor. Shall we revisit it later? I think someone is depending on this PR to debug...

Reply:
Don't think this is hard to fix... How about simply pass `old_rows.len() + new_rows.len()` as capacity?

Reply:
Working on a quick fix 🥵